plugins/auth-backend/src/lib/OAuthProvider.ts

```diff
@@ -39,9 +39,41 @@ export type Options = {
   tokenIssuer: TokenIssuer;
 };
 
+/* Return the value of `env` key, if it is exists, encoded within
+     the `state` parameter in the request
+  */
+const readStateParameter = (
```

Comment:
Not sure about all of this manual string fiddling - can't the state be base64 of the json encoded state object or something, and make a proper Typescript type for the state so we can use it more cleanly?

Reply:
Yes, I think turning the state into a type makes sense. 
I admit that the serializing is messy.  I am just getting started with Typescript and my code wont be idiomatic. Will try to find a simple way to serialize and deserialize it.